content/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,368 @@
+
+```markdown
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: otel-deployment-kubernetes-monitor
+  labels:
+    app: otel-kubernetes-monitor
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: otel-kubernetes-monitor
+  template:
+    metadata:
+      labels:
+        app: otel-kubernetes-monitor
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: otel-sa-kubernetes-monitor
+      containers:
+        - name: otel-kubernetes-monitor
+          image: otel/opentelemetry-collector:0.50.0
+          command:
+            - "/otelcol"
+            - "--config=/conf/otel-collector-config.yaml"
+          volumeMounts:
+            - name: otel-collector-config-vol-kubernetes-monitor
+              mountPath: /conf
+      volumes:
+        - name: otel-collector-config-vol-kubernetes-monitor
+          configMap:
+            name: otel-collector-conf-kubernetes-monitor
+            items:
+              - key: otel-collector-config
+                path: otel-collector-config.yaml
+```
+
+##### 3. Deploy the OpenTelemetry collector:
+
+```markdown
+$ kubectl apply -f otel-collector-config.yaml -n skywalking
+$ kubectl apply -f otel-collector-deploy.yaml -n skywalking
+```
+
+```markdown
+$ kubectl  get all -n skywalking -l app=otel-kubernetes-monitor
+NAME                                                      READY   STATUS    RESTARTS   AGE
+pod/otel-deployment-kubernetes-monitor-798cdd8486-gz885   1/1     Running   0          93d
+
+NAME                                                 READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/otel-deployment-kubernetes-monitor   1/1     1            1           96d
+
+NAME                                                            DESIRED   CURRENT   READY   AGE
+replicaset.apps/otel-deployment-kubernetes-monitor-798cdd8486   1         1         1       96d
+```
+
+After all setup steps are finished, the Kubernetes metrics will be showing up in SkyWalking UI, under *Kubernetes* tab:
+
+![](/img/sw-k8s-clu.png)
+
+You can check the﻿ Kubernetes overview from SkyWalking UI:
+
+![](/img/sw-k8s-overview.png)
+
+And the Kubernetes nodes:  
+
+![](/img/sw-k8s-node.png)
+
+The Kubernetes worker node:
+
+![](/img/sw-k8s-node-instance.png)
+
+And the  Kubernetes services:
+
+![](/img/sw-k8s-svc.png)
+
+## Conclusion
+
+In this blog post, I discussed the challenges in Kubernetes monitoring and why it’s important in Kubernetes monitoring in HPE GreenLake for Private Cloud Enterprise. I then took the Apache SkyWalking as the application performance monitoring (APM) tool and showed the detailed process of setting it up, as a *self-managed* environment in HPE GreenLake for Private Cloud Enterprise, for monitoring Kubernetes cluster. It provides a way to gain the visibility of the objects and operations performed by Kubernetes, and to resolve issues in the cluster.
```

Comment:
Change the second part of the first sentence to "...and why it's important **for** Kubernetes monitoring..."

I really didn't see much of a discussion about the challenges in Kubernetes monitoring. Did you want to get more into that?

In the second half of the second sentence, take off the comma after "Private Cloud Enterprise" and add the word "a" prior to "Kubernetes cluster."

Reply:
Yes, I've added more discussion to Introduction section about the challenges in K8s monitoring, but trying to keep it not too long. Please review it and let me know what you think.